applications/tari_console_wallet/src/ui/state/tasks.rs

```diff
@@ -215,3 +225,101 @@ pub async fn send_one_sided_to_stealth_address_transaction(
+                .unwrap();
+        },
+        Ok((burn_tx_id, original_proof)) => {
+            loop {
+                match event_stream.recv().await {
+                    Ok(event) => {
+                        if let TransactionEvent::TransactionCompletedImmediately(completed_tx_id) = &*event {
+                            if burn_tx_id == *completed_tx_id {
+                                let wrapped_proof = BurntProofBase64 {
+                                    reciprocal_claim_public_key: original_proof.reciprocal_claim_public_key.to_vec(),
+                                    commitment: original_proof.commitment.to_vec(),
+                                    ownership_proof: original_proof.ownership_proof.map(|x| {
+                                        CommitmentSignatureBase64 {
+                                            public_nonce: x.public_nonce().to_vec(),
+                                            u: x.u().to_vec(),
+                                            v: x.v().to_vec(),
+                                        }
+                                    }),
+                                    range_proof: original_proof.range_proof.0,
+                                };
+
+                                let serialized_proof = serde_json::to_string_pretty(&wrapped_proof)
+                                    .expect("failed to serialize burn proof");
+
+                                let filepath = burn_proof_filepath
+                                    .unwrap_or_else(|| PathBuf::from(format!("{}.json", burn_tx_id.as_u64())));
+
+                                std::fs::write(filepath, serialized_proof.as_bytes())
+                                    .expect("failed to save burn proof");
+
+                                let proof_id = random::<i32>().abs();
+                                let ts = Utc::now().naive_utc();
+
+                                if let Err(e) = db.insert_burnt_proof(
+                                    proof_id,
+                                    original_proof.reciprocal_claim_public_key.to_hex(),
+                                    serialized_proof.clone(),
+                                    ts,
+                                ) {
+                                    error!("failed to save burnt proof to the database: {:#?}", e);
+                                    return;
+                                }
+
+                                result_tx
+                                    .send(UiTransactionBurnStatus::TransactionComplete((
+                                        proof_id,
+                                        original_proof.reciprocal_claim_public_key.to_hex(),
+                                        serialized_proof,
+                                        ts,
+                                    )))
+                                    .unwrap();
+
+                                return;
+                            }
+                        }
```

Comment:
I would add an `else` here because there could be other events that are getting swallowed, potentially introducing a bug in future